File: 2744/go/main.go
func maximumNumberOfStringPairs(words []string) int {
	m := make(map[string]int)
	var rd string
	var ans int
	for _, word := range words {
		rd = string(word[1]) + string(word[0])
		if m[rd] == 1 {
			ans++
		}
		m[word] = 1
	}

	return ans
}

File: 2744/python3/main.py
class Solution:
    def maximumNumberOfStringPairs(self, words: list[str]) -> int:
        m = {}
        result = 0
        for word in words:
            rd = word[1] + word[0]
            if m.get(rd) == 1:
                result += 1
            m[word] = 1
        return result
